opm/simulators/wells/WellGroupHelpers.cpp

```diff
@@ -385,6 +410,12 @@ namespace WellGroupHelpers
                     continue;
 
                 int well_index = it->second[0];
+
+                if (! wellIsOwned(well_index, wellName, wellState) ) // Only sum once
```

Comment:
Unfortunately, this function was not deleted from the cpp file when it was moved to the hpp file due to becoming a template. I think you need to make the change there instead. You can remove the commented-out function for future sanity preservation!

However, your new code in WellState.hpp made me think that it can be possible to have this function in the cpp file anyway, by not using a template argument for the Comm. I leave to you to make a decision on this for now.